### PR TITLE
Add a mechanism to generate README.md (for table of CHIPs)

### PR DESCRIPTION
**Pull request by mppf**
Creates README.md.in and an update-readme.bash that generates the table of CHIPS.

### CODE REVIEW

**Comment by mppf** [comment]
@lydia-duncan, can you review this?

File: doc/chips/README.md.in [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Chapel Improvement Proposals (CHIPs)
+
+See the first CHIP for a an overview.
```
Comment (lydia-duncan) [inline doc/chips/README.md.in]: Probably just want "an" here instead of "a an"

**Comment by lydia-duncan** [comment]
I think fixing the typo may cause README.md to need to be updated as well.  I'm assuming you still want that file committed to the repository (because it is easier to traverse that way).  This looks good to me otherwise :+1:

**Comment by mppf** [comment]
Thanks. I'm planning to add some other CHIPs soon, so I'll update README.md when I do that.

**Pull request merged**